File: backend/src/main/java/com/jbconnect/healthclinix/entity/user/mapper/UserMapper.java
package com.jbconnect.healthclinix.entity.user.mapper;

import java.util.List;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import com.jbconnect.healthclinix.entity.user.dto.UserRequestDTO;
import com.jbconnect.healthclinix.entity.user.dto.UserResponseDTO;
import com.jbconnect.healthclinix.entity.user.model.UserModel;

@Mapper(componentModel = "spring")
public interface UserMapper {

	@Mapping(source = "username", target = "userName")
    UserResponseDTO userModelToUserResponseDTO(UserModel userModel);

    UserModel userRequestToUserModel(UserRequestDTO userRequestDTO);

    @Mapping(source = "username", target = "userName")
    List<UserResponseDTO> userModelListToUserResponseDTOList(List<UserModel> userModelList);

    UserRequestDTO userModelToUserRequestDTO(UserModel userModel);
}

File: backendkotlin/src/main/kotlin/com/jbconnect/healthclinix/healthclinix/entity/user/mapper/UserMapper.kt
package com.jbconnect.healthclinix.healthclinix.entity.user.mapper

import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserRequestDTO
import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserResponseDTO
import com.jbconnect.healthclinix.healthclinix.entity.user.model.UserModel
import org.mapstruct.Mapper
import org.mapstruct.Mapping


@Mapper(componentModel = "spring")
interface UserMapper {
    @Mapping(source = "username", target = "userName")
    fun userModelToUserResponseDTO(userModel: UserModel): UserResponseDTO

    fun userRequestToUserModel(userRequestDTO: UserRequestDTO): UserModel

    @Mapping(source = "username", target = "userName")
    fun userModelListToUserResponseDTOList(userModelList: List<UserModel?>): List<UserResponseDTO?>

    fun userModelToUserRequestDTO(userModel: UserModel): UserRequestDTO
}